storage/src/main/java/com/tencent/rss/storage/handler/impl/HdfsClientReadHandler.java

```diff
@@ -51,7 +66,96 @@ public HdfsClientReadHandler(
         ShuffleStorageUtils.getShuffleDataPathWithRange(appId,
             shuffleId, partitionId, partitionNumPerRange, partitionNum));
     init(fullShufflePath);
-    readAllIndexSegments();
+    readerIterator = indexReaderMap
+        .entrySet()
+        .stream()
+        .sorted(Comparator.comparing(Entry::getKey))
+        .collect(Collectors.toList())
+        .iterator();
+  }
+
+  public ShuffleIndexResult readShuffleIndex() {
+    if (!readerIterator.hasNext()) {
+      return new ShuffleIndexResult();
+    }
+
+    Entry<String, HdfsFileReader> entry = readerIterator.next();
+    currentPath = entry.getKey();
+    long start = System.currentTimeMillis();
+    HdfsFileReader reader = entry.getValue();
+    if (reader == null) {
+      LOG.error("Reader of {} is invalid!", currentPath);
+      return new ShuffleIndexResult();
+    }
+
+    try {
+      byte[] indexData = reader.readAllData();
```

Comment:
why do we use readAllData instead of readIndex?

Reply:
The same as above